runtime/src/main/java/io/quarkiverse/asyncapi/config/channels/ChannelConfigurerFactory.java

```diff
@@ -0,0 +1,20 @@
+package io.quarkiverse.asyncapi.config.channels;
+
+import java.util.Map;
+import java.util.Objects;
+
+import com.asyncapi.v2.model.server.Server;
+
+public class ChannelConfigurerFactory {
+
+    private ChannelConfigurerFactory() {
+    }
+
+    // TODO make this extensible
+    private static Map<String, ChannelConfigurer> configurers = Map.of("http", new HttpChannelConfigurer(), "kafka",
```

Comment:
Besides making this extensible, we should add more configurers

Reply:
Maybe using a ServiceLoader here?

Reply:
Yes, thats the idea, but I feel it was too much for a single PR

Reply:
@gastaldi Or mayby it wasnt ;)
ChannelConfigurer implemented as serviceloader